File: Game.Application/GameJS/src/utils/specifications/NotSpecification.ts
import { ISpecification } from "./ISpecification";
export class NotSpecification<TSituation> implements ISpecification<TSituation, boolean> {
  private readonly original: ISpecification<TSituation, boolean>;
  constructor(original: ISpecification<TSituation, boolean>) {
    this.original = original;
  }
  execute(situation: TSituation): boolean {
    return !this.original.execute(situation);
  }
}

File: Game.Domain/Specifications/NotSpecification.cs
namespace Game.Domain.Specifications
{
    public class NotSpecification<TSituation> : ISpecification<TSituation, bool>
    {
        private readonly ISpecification<TSituation, bool> original;

        public NotSpecification(ISpecification<TSituation, bool> original)
        {
            this.original = original;
        }

        public bool Execute(TSituation situation)
        {
            return !original.Execute(situation);
        }
    }

}
